core/packages/contracts/src/utils/Bitfield.sol

```diff
@@ -97,27 +103,27 @@ library Bitfield {
                 x = (x & M64) + ((x >> 64) & M64); //put count of each 128 bits into those 128 bits
                 x = (x & M128) + ((x >> 128) & M128); //put count of each 256 bits into those 256 bits
                 count += x;
+                if (count >= threshold) {
+                    break;
+                }
             }
-            return count;
         }
+        return count >= threshold;
```

Comment:
@doubledup I did some more tests here for gas usage, a bit weird but seems the previous `countSetBits` is the most efficient, so I revert the optimization here. Could you please check if this is also the case in your setup?

Reply:
huh, that's a bit surprising but I think I've found why. It could be that `vset.length` [is 300 in our test data](https://github.com/Snowfork/snowbridge/blob/main/core/packages/contracts/scripts/ffiWrapper.ts#L12), so `bitfield.length` is 2. Since we're looking for 2/3 of bits being set, we need to run both iterations. But now we check an extra condition in each iteration, which makes it more expensive.

300 seems representative of mainnet atm so let's shelve this for now. We can only expect this optimisation to help with >= 1024 validators:
We're looking for >2/3 of bits set in chunks of 256. So even at 768 validators we can only have 2/3 after 2 iterations and have to run the 3rd iteration.

Reply:
Yes, that make sense!

Reply:
Just realised I made a mistake here, it can help when we have >768 validators. Checking the first 768 validator positions in the bitmask can give us 2/3 of validators for <1152 validators.

This is still only something to keep tabs on as Polkadot's validator count [approaches 1000](https://wiki.polkadot.network/docs/faq#why-will-polkadot-have-only-1000-validators-while-other-projects-have-hundreds-of-thousands), since there's still no guarantee that this will be a net benefit given the extra checks in each loop.